action/protocol/staking/protocol.go

```diff
@@ -476,7 +476,11 @@ func (p *Protocol) ActiveCandidates(ctx context.Context, sr protocol.StateReader
 	featureCtx := protocol.MustGetFeatureCtx(ctx)
 	for i := range list {
 		if p.contractStakingIndexer != nil && featureCtx.AddContractStakingVotes {
-			list[i].Votes.Add(list[i].Votes, p.contractStakingIndexer.CandidateVotes(list[i].Owner))
+			contractVotes, err := p.contractStakingIndexer.CandidateVotes(list[i].Owner, height)
```

Comment:
`height` here equals to the height of block to be committed, that is, current factory/indexer's height + 1
so should use `height-1`? pls confirm

Reply:
should be covered by unit test

Reply:
by using `sr.Height() - 1` instead of the input height, the original logic is preserved, ensuring the retrieval of data at the latest height in the State, rather than a specific height